packages/service/src/service.ts

```diff
@@ -56,6 +65,20 @@ app.use(passport.session());
 
 app.post("/login", passport.authenticate("local", { successRedirect: "/" }));
 
+app.post("/user/update_password", async function (req, resp) {
+  await updatePassword(req.body);
```

Comment:
the problem with the requests blowing up is from here. All of these zapatos promises can fail, and if/when then do, the response is never triggered. We should catch errors and handle them.

`401 unauthorized` seems appropriate.

Select users would be a `404` I guess? What do you think?

Reply:
Since these are server errors, wouldn't `500 Internal Server Error` be more appropriate?

Reply:
@tmsavage `500 Internal Server Error` is what is sent to the client when there is an unhandled exception and is pretty ambiguous - instead we want to handle server errors, and then send the client a more informative status code

Reply:
Here is some documentation about how express handles errors, pay special attention to the async sections

https://expressjs.com/en/guide/error-handling.html